Zend/zend_inheritance.c

```diff
@@ -1561,6 +1561,36 @@ ZEND_API void zend_do_inheritance_ex(zend_class_entry *ce, zend_class_entry *par
 }
 /* }}} */
 
+static zend_always_inline bool check_trait_property_or_constant_value_compatibility(zend_class_entry *ce, zval *op1, zval *op2) /* {{{ */
+{
+	bool is_compatible;
+	zval op1_tmp, op2_tmp;
+
+	/* if any of the values is a constant, we try to resolve it */
+	if (UNEXPECTED(Z_TYPE_P(op1) == IS_CONSTANT_AST)) {
+		ZVAL_COPY_OR_DUP(&op1_tmp, op1);
+		zval_update_constant_ex(&op1_tmp, ce);
+		op1 = &op1_tmp;
+	}
+	if (UNEXPECTED(Z_TYPE_P(op2) == IS_CONSTANT_AST)) {
+		ZVAL_COPY_OR_DUP(&op2_tmp, op2);
+		zval_update_constant_ex(&op2_tmp, ce);
+		op2 = &op2_tmp;
+	}
+
+	is_compatible = !fast_is_not_identical_function(op1, op2);
```

Comment:
Question: I understand this comes from the "inversion" of originally `not_compatible` to now `is_compatible`, but resulting in a "double negation"; would the following be "better"?
```suggestion
	is_compatible = fast_is_identical_function(op1, op2);
```

Reply:
Definitely, thanks!